vg/font_test.go

```diff
@@ -0,0 +1,213 @@
+					Descent: -2.595703125,
+					Height:  13.798828125,
+				},
+				24: FontExtents{
+					Ascent:  21.38671875,
+					Descent: -5.19140625,
+					Height:  27.59765625,
+				},
+			},
+		},
+		{
+			font: "Times-Bold",
+			want: map[Length]FontExtents{
+				10: FontExtents{
+					Ascent:  8.9111328125,
+					Descent: -2.1630859375,
+					Height:  11.4990234375,
+				},
+				12: FontExtents{
+					Ascent:  10.693359375,
+					Descent: -2.595703125,
+					Height:  13.798828125,
+				},
+				24: FontExtents{
+					Ascent:  21.38671875,
+					Descent: -5.19140625,
+					Height:  27.59765625,
+				},
+			},
+		},
+		{
+			font: "Times-Italic",
+			want: map[Length]FontExtents{
+				10: FontExtents{
+					Ascent:  8.9111328125,
+					Descent: -2.1630859375,
+					Height:  11.4990234375,
+				},
+				12: FontExtents{
+					Ascent:  10.693359375,
+					Descent: -2.595703125,
+					Height:  13.798828125,
+				},
+				24: FontExtents{
+					Ascent:  21.38671875,
+					Descent: -5.19140625,
+					Height:  27.59765625,
+				},
+			},
+		},
+		{
+			font: "Times-BoldItalic",
+			want: map[Length]FontExtents{
+				10: FontExtents{
+					Ascent:  8.9111328125,
+					Descent: -2.1630859375,
+					Height:  11.4990234375,
+				},
+				12: FontExtents{
+					Ascent:  10.693359375,
+					Descent: -2.595703125,
+					Height:  13.798828125,
+				},
+				24: FontExtents{
+					Ascent:  21.38671875,
+					Descent: -5.19140625,
+					Height:  27.59765625,
+				},
+			},
+		},
+	} {
+		for _, size := range []Length{10, 12, 24} {
+			fnt, err := MakeFont(tc.font, size)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			got := fnt.Extents()
+			if got, want := got, tc.want[size]; got != want {
+				t.Errorf(
+					"invalid font extents for %q, size=%v:\ngot= %#v\nwant=%#v",
+					tc.font, size, got, want,
+				)
+			}
+		}
+	}
+}
+
+func TestFontWidth(t *testing.T) {
+	fnt, err := MakeFont("Times-Roman", 12)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, tc := range []struct {
+		txt  string
+		want Length
+	}{
+		{" ", 3},
+		{"i", 3.333984375},
+		{"q", 6},
+		{"A", 8.666015625},
+		{"F", 6.673828125},
+		{"T", 7.330078125},
+		{"V", 8.666015625},
+		{"Δ", 7.716796875},
+		{"∇", 9.333984375},
+		{"  ", 6},
+		{"AA", 17.33203125},
+		{"Aq", 14.666015625},
+		{"Fi", 10.0078125},
+		{"Ti", 10.2421875},
+		{"AV", 15.78515625},
+		{"A∇", 18},
+		{"VA", 15.78515625},
+		{"VΔ", 16.3828125},
+		{"∇Δ", 17.05078125},
+		{"   ", 9},
+		{"T T", 17.2265625},
+	} {
+		t.Run(tc.txt, func(t *testing.T) {
+			got := fnt.Width(tc.txt)
+			if got, want := got, tc.want; got != want {
+				t.Fatalf(
+					"invalid width: got=%v, want=%v",
+					got, want,
+				)
+			}
+		})
+	}
+}
+
+func TestFontKern(t *testing.T) {
+	fnt, err := MakeFont("Times-Roman", 12)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, tc := range []struct {
+		txt  string
+		want fixed.Int26_6
+	}{
+		{"AV", -264},
+		{"A∇", 0},
+		{"AΔ", 0},
+		{"AA", 0},
+		{"VA", -264},
+		{"∇A", 0},
+		{"∇Δ", 0},
```

Comment:
Here too; some are surprising for a non-professional-typographer.

Reply:
I'll add a comment (mainly to note where (I think) a Kerning should be had but isn't. probably a shortcoming of the font definition itself rather than the package that decoded it, though)